Source/OrchBase/Composer.cpp

```diff
@@ -1965,76 +2646,93 @@ void Composer::writeThreadContextInitialiser(P_thread* thread, DevT_t* devT,
+        * required.
+        */
+        buffCount = 1;     // Intentionally 1 to cope with wrapping.
 
-    if (outTypCnt)  // If we have output pins
-    {               // Iterate through devices counting connected output pins.
-        WALKLIST(DevI_t*, placer->threadToDevices.at(thread), dev)
-        {
-            WALKVECTOR(PinT_t*, devT->PinTO_v, pin)
+        if(devT->pPinTSO)
+        {   // There is a supervisor output, increase output count by dev count.
+            buffCount += numberOfDevices;
+        }
+
+        if (outTypCnt) // If we have output pins
+        {              // Iterate through devices counting connected output pins
+            WALKLIST(DevI_t*, placer->threadToDevices.at(thread), dev)
             {
-                // Check that we have a connection
-                std::map<std::string,PinI_t *>::iterator pinSrch;
-                pinSrch = (*dev)->Pmap.find(devT->Name());
-                if(pinSrch != (*dev)->Pmap.end())
+                WALKVECTOR(PinT_t*, devT->PinTO_v, pin)
                 {
-                    if(pinSrch->second->Key_v.size())
+                    // Check that we have a connection
+                    std::map<std::string,PinI_t *>::iterator pinSrch;
+                    pinSrch = (*dev)->Pmap.find(devT->Name());
+                    if(pinSrch != (*dev)->Pmap.end())
                     {
-                        outputCount++;
+                        if(pinSrch->second->Key_v.size())
+                        {
+                            buffCount++;
 
-                        // If 0, we have overflowed & there is no point cont
-                        if (outputCount==0) break;
+                            // If 0, we have overflowed & there is no point cont
+                            if (buffCount==0) break;
+                        }
                     }
                 }
-            }
-            if (outputCount==0)
-            {
-                outputCount = MAX_RTSBUFFSIZE;
-                break;
+                if (buffCount==0)
+                {
+                    buffCount = builderGraphI->rtsBuffSizeMax;
+                    break;
+                }
             }
         }
-    }
 
-    if (outputCount > MAX_RTSBUFFSIZE)
-    { // If we have too many pins for one buffer entry per ping, set to max &warn.
-    // This may need a check adding to the Softswitch to stop buffer overflow.
-        outputCount = MAX_RTSBUFFSIZE;
-        //TODO: Barf
-        //par->Post(819,int2str(thread_num),int2str(coreNum),
-        //            int2str(MAX_RTSBUFFSIZE), int2str(MAX_RTSBUFFSIZE));
-    }
-    else if (outputCount < MIN_RTSBUFFSIZE)
-    {
-        outputCount = MIN_RTSBUFFSIZE;
+        if (buffCount > builderGraphI->rtsBuffSizeMax)
+        { // If we have too many pins for one buffer entry per ping, set to max &warn.
+        // This may need a check adding to the Softswitch to stop buffer overflow.
```

Comment:
If not post, you can always write a `#warning` into the generated code.

Reply:
That is a good point. Added.